[Shi Wen] iP

shwene opened the pull request:
## Duke 💯 

> “Everything has beauty, but not everyone can see.” — [Confucius](https://youtu.be/og1Pbn8OufI)

### Duke helps you store whichever tasks needs be. It's

- [x] Free
- [x] **Fun**
- [x] Fantastic

#### To use:
1. Register [here](https://youtu.be/Nrk8sqZfsgI)
2. Download the installer
3. *Run*

#### Disclaimer:
- The creator assumes no responsibility or liability for any errors or failures
- You agree and acknowledge that Duke is entitled to modify, improve or discontinue any of its services at its sole discretion

#### Here's a peak at the `main` method:

```java
public static void main(String[] args) {
    new Duke("data/tasks.txt").run();
}
```

Commit:
Add Gradle support

shwene committed:
Completed level1

shwene committed:
added level2

shwene committed:
Completed level3

shwene committed:
Completed level4

shwene committed:
Completed level5

shwene committed:
Completed level6

shwene committed:
Updated testcase & ran automated I/O testing

shwene committed:
Refactor code to OOP style

shwene committed:
Add all files to package duke

shwene committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

shwene committed:
Integrate gradle with code

shwene committed:
Implement JUnit tests

shwene committed:
Update task list fail message

shwene committed:
Remove unnecessary import from Deadline class

shwene committed:
Merge branch 'branch-Level-7'

shwene committed:
Merge branch 'branch-Level-8'

shwene committed:
Add javadocs to public methods and classes

shwene committed:
Update coding style to comply with standards

shwene committed:
Add find operation to Duke

shwene committed:
Merge branch 'branch-A-JavaDoc'

shwene committed:
Merge branch 'branch-A-CodingStandard'

shwene committed:
Merge branch 'branch-Level-9'

beetee17 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,79 @@
+                    ui.deleteTaskConfirmation(deletedTask, tasks.getSize());
+                }
+            }
+        }
```
Would it be possible to clean this up and avoid nested if-else statements?

beetee17 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,51 @@
+        if (index > 0) {
+            return desc.substring(index + 1);
+        }
+        return "OTOT";
```
You might want to consider extracting this into a static final variable, instead of using it inline as a "magic value". This would also help you avoid future refactoring if used in multiple places.

beetee17 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,66 @@
+    /**
+     * Displays linebreak to users.
+     */
+    protected void lineBreak() {
```
Nice job on following all the naming conventions throughout the PR! Just in this case you might want to use a verb to better specify the method's action e.g. `displayLineBreak`

beetee17 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,104 @@
+        if (!isKeyword(keyword) || keyword.isBlank()) {
+            throw new DukeException("I'm sorry, but I don't know what that means :-(");
+        }
+    }
```
I initially thought to suggest changing the return type to a boolean due to the method name. On second thought, this seems valid, but I would suggest renaming the methods to something like `checkIsValidTask` or `validateTaskDescriptionIsNotEmpty`

beetee17 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,50 @@
+        while (s.hasNext()) {
+            String taskText = s.nextLine();
+            char taskType = taskText.charAt(3);
+            if (taskType == 'T') {
+                tasks.add(new Todo(taskText));
+            } else if (taskType == 'D') {
+                tasks.add(new Deadline(taskText));
+            } else if (taskType == 'E') {
+                tasks.add(new Event(taskText));
+            }
```
Consider extracting task type identifiers into a class field, e.g. `Todo.TYPE_IDENTIFIER`

beetee17 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,48 @@
+
+    private void run() {
+        ui.intro();
+        //int id = 0;
```
I think you forgot to remove this commented-out piece of code!

shwene replied:
Thanks for all the pointers!

beetee17 reviewed (commented):
Great job overall regarding code quality! Just a few issues here and there that are easily fixable. Keep it up!

jeromepui commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,82 @@
+        return output.toString();
+    }
+
+    protected String find(String term) {
```
`find` seems a little vague. Perhaps you could rename it to `findTasks` instead.
```suggestion
    protected String findTasks(String term) {
```

jeromepui commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,82 @@
+     * @return Task updated as incomplete.
+     */
+    protected Task unmarkTask(int index) {
+        Task undidTask = tasks.get(index).undoTask();
```
`undidTask` doesn't seem to be standard English. Perhaps you could rename it to `incompleteTask`.
```suggestion
        Task incompleteTask = tasks.get(index).undoTask();
```

jeromepui commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,79 @@
+     * Check whether a given Command instance is consists of the keyword bye (non case-sensitive).
+     * @return true if the command instance is a bye command, false otherwise.
+     */
+    protected boolean isBye(){
```
`isBye` seems vague. Perhaps you could rename it to `isByeCommand`.
```suggestion
    protected boolean isByeCommand(){
```

jeromepui commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,66 @@
+    /**
+     * Displays welcome message to users.
+     */
+    protected void intro() {
```
`intro` doesn't seem suitable for a method name. Perhaps you could rename it to `greet`.
```suggestion
    protected void greet() {
```

shwene replied:
Thanks for all the pointers!

jeromepui reviewed (commented):
LGTM. 👍 Just need to fix some minor naming issues.

shwene committed:
Add a GUI to duke

shwene committed:
Merge branch 'branch-Level-10'

shwene committed:
Add ancillary classes to support Command class

shwene committed:
Refactor Parser class with streams

shwene committed:
Add assertion to subclasses of the Task class

The assertions ensure that the various task objects are initalised correctly as per accordance to the label of their task type so that the task type label of tasks is shown accurately to the user.

shwene committed:
Refactor multiple classes to improve code quality

- Remove excess code after the addition of GUI to Duke
- Replace magic literals with static final constants
- Rename method names to be more human readable

shwene committed:
Merge pull request #1 from shwene/branch-A-CodeQuality

Refactor multiple classes to improve code quality

shwene committed:
Merge branch 'master' of https://github.com/shwene/ip

shwene committed:
Merge branch 'branch-A-Streams'

shwene committed:
Merge branch 'branch-A-Assertions'

shwene committed:
Update GUI of Duke

shwene committed:
Update Javadocs to most public and protected methods

shwene committed:
Update README.md

shwene committed:
Clean task log

shwene committed:
Update README.md

shwene committed:
Update README.md

shwene committed:
Update Javadocs to adhere to convention

shwene committed:
Update FileWriter to save to relative path

shwene committed:
Update help command content

shwene committed:
Update README.md

shwene committed:
Update README.md

shwene committed:
Update README.md

shwene committed:
Update README.md

shwene committed:
Update README.md

shwene committed:
Update README.md
